events/bingo/webApps/bingo.js

```diff
@@ -348,16 +379,127 @@ function replaceCalledNumbers(newCalledNumbers, response) {
             text: "replaceCalledNumbers was called, but there was no currentTableName!"
         };
 
-        response.statusCode = 200;
+        response.statusCode = 500;
         response.setHeader('Content-Type', 'application/json');
         return response.end(JSON.stringify(responseObject));
     }
 }
 
+
+// Given some input data, formats the human-readable text data
+// in an arbitrary way as needed by the team.
+function formatDataForExport(tablePrefix, allWinnersReceive, data, response) {
+    var finalResponseText = `Requested Table Prefix: "${tablePrefix}"\n\n`;
+
+    for (var i = 0; i < data.length; i++) {
+        finalResponseText += "Bingo Round #" + (i + 1) + "\n";
+
+        var currentUsernameList = data[i];
+        var someoneWonAllPlayersWin = false;
+        var currentResponseText = "";
+        for (var j = 0; j < currentUsernameList.length; j++) {
+            currentResponseText += allWinnersReceive + ",";
+            currentResponseText += "Event Grant,";
+            currentResponseText += currentUsernameList[j].username;
+            currentResponseText += "\n";
+
+            if (String(currentUsernameList[j].prizeWon).toLowerCase().indexOf("all") > -1) {
+                someoneWonAllPlayersWin = true;
+            }
+        }
+
+        if (someoneWonAllPlayersWin) {
+            finalResponseText += "Amount of HFC,Type,Username\n";
+            finalResponseText += currentResponseText;
+        } else {
+            finalResponseText += "NOBODY WON 'All Players Win' in this round.";
+        }
+
+        finalResponseText += "\n\n\n";
+    }
+
+    response.statusCode = 200;
+    response.setHeader('Content-Type', 'text');
+    return response.end(finalResponseText);
+}
+
+// Handles data export request as per the user-submitted form.
+// Uses password-based authentication using the same password
+// as the db password.
+function handleDataExportRequest(formData, response) {
+    if (formData.password !== dbInfo.mySQLPassword) {
+        var responseObject = {
+            status: "error",
+            text: "Not authorized."
+        };
+    
+        response.statusCode = 401;
+        response.setHeader('Content-Type', 'application/json');
+        return response.end(JSON.stringify(responseObject));    
+    }
+
+    var query = `SHOW TABLES LIKE '${formData.prefix}_%'`;

```

Comment:
I looked this one up btw 
_ = match one char
% = match any number of chars
%cat% = blackcattail
so I don't think you need the _ 
 
​

Reply:
you're right that i don't NEED the semicolon, but it is more completely correct. table names are formatted `<prefix>_<timestamp>`